sshd-common/src/main/java/org/apache/sshd/common/channel/PtyChannelConfiguration.java

```diff
@@ -94,6 +95,8 @@ public Map<PtyMode, Integer> getPtyModes() {
 
     @Override
     public void setPtyModes(Map<PtyMode, Integer> ptyModes) {
+        ptyModes.forEach(
+            (key, value) -> BufferUtils.validateUint32Value(value, "Mode %s has invalid value %s", key, value));
```

Comment:
Make a copy of the map? The caller might add more values afterwards... And should getPtyModes() return a read-only copy?